src/Internal/Service/PaymentRequestService.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Class PaymentRequestService
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Service;
+
+use WC_Payments_API_Payment_Intention;
+use WCPay\Core\Server\Request\Create_And_Confirm_Intention;
+use WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception;
+use WCPay\Core\Exceptions\Server\Request\Extend_Request_Exception;
+use WCPay\Core\Exceptions\Server\Request\Immutable_Parameter_Exception;
+use WCPay\Internal\Payment\PaymentContext;
+
+/**
+ * Service for managing requests, connecting payments and the API.
+ */
+class PaymentRequestService {
+	/**
+	 * Creates a payment intent. To be replaced by an abstraction soon.
+	 *
+	 * @param PaymentContext $context Context for the payment.
+	 * @return WC_Payments_API_Payment_Intention
+	 * @throws Invalid_Request_Parameter_Exception
+	 * @throws Extend_Request_Exception
+	 * @throws Immutable_Parameter_Exception
+	 */
+	public function create_intent( PaymentContext $context ) {
+		$request = Create_And_Confirm_Intention::create();
+		$request->set_amount( $context->get_amount() );
+		$request->set_currency_code( $context->get_currency() );
+		$request->set_payment_method( $context->get_payment_method()->get_id() );
+		$request->set_customer( $context->get_customer_id() );
+		$request->set_capture_method( $context->should_capture_manually() );
+		$request->set_metadata( $context->get_metadata() );
+		$request->set_level3( $context->get_level3_data() );
+		$request->set_payment_methods( [ 'card' ] ); // Initial payment process only supports cards.
+		$request->set_cvc_confirmation( $context->get_cvc_confirmation() );
+
+		$fingerprint = $context->get_fingerprint();
+		if ( ! is_null( $fingerprint ) ) {
+			$request->set_fingerprint( $fingerprint );
+		}
```

Comment:
```suggestion
		$request->set_fingerprint( $context->get_fingerprint() ?? '' );

```

I think we would fallback to empty string `''`. Check other usages of fingerprint in the old gateway classes: 

- https://github.com/Automattic/woocommerce-payments/blob/f2412c1eb6c18f43ad7e22c547998fa4fe53a886/includes/class-wc-payment-gateway-wcpay.php#L1277
- https://github.com/Automattic/woocommerce-payments/blob/f2412c1eb6c18f43ad7e22c547998fa4fe53a886/includes/payment-methods/class-upe-payment-gateway.php#L360-L360

Reply:
Wow, this one was too easy to underestimate. I thought that we were passing an empty string just to avoid `if()`s down the payment process, but it turns out that even with an empty string, we need to call `set_fingerprint()` because it is the link between the request, and the `Buyer_Fingerprinting_Service` class. See https://github.com/Automattic/woocommerce-payments/blob/rpp/basic-process/includes/core/server/request/trait-intention.php#L23 and https://github.com/Automattic/woocommerce-payments/blob/rpp/basic-process/includes/core/server/request/trait-intention.php#L24.

Defaulting to an empty string shortly...

Reply:
Fixed in https://github.com/Automattic/woocommerce-payments/pull/7332/commits/8eb8bc11f30daec65e10614b7b511e6f32407885